common/address_test.go

```diff
@@ -23,20 +23,41 @@ func TestAddress(t *testing.T) {
 
 func TestDecodeBtcAddress(t *testing.T) {
 	t.Run("invalid string", func(t *testing.T) {
-		_, err := DecodeBtcAddress("�U�ڷ���i߭����꿚�l", 18332)
+		_, err := DecodeBtcAddress("�U�ڷ���i߭����꿚�l", BtcTestNetChain().ChainId)
 		require.ErrorContains(t, err, "runtime error: index out of range")
 	})
 	t.Run("invalid chain", func(t *testing.T) {
 		_, err := DecodeBtcAddress("14CEjTd5ci3228J45GdnGeUKLSSeCWUQxK", 0)
-		require.ErrorContains(t, err, "is not a Bitcoin chain")
+		require.ErrorContains(t, err, "is not a bitcoin chain")
 	})
-	t.Run("nil pointer dereference", func(t *testing.T) {
-		_, err := DecodeBtcAddress("tb1qy9pqmk2pd9sv63g27jt8r657wy0d9uee4x2dt2", 18332)
-		require.ErrorContains(t, err, "runtime error: invalid memory address or nil pointer dereference")
+	t.Run("invalid checksum", func(t *testing.T) {
+		_, err := DecodeBtcAddress("tb1qy9pqmk2pd9sv63g27jt8r657wy0d9uee4x2dt2", BtcTestNetChain().ChainId)
+		require.ErrorContains(t, err, "invalid checksum")
 	})
-	t.Run("valid address", func(t *testing.T) {
-		_, err := DecodeBtcAddress("bcrt1qy9pqmk2pd9sv63g27jt8r657wy0d9uee4x2dt2", 18444)
+	t.Run("valid legacy main-net address address incorrect params TestNet", func(t *testing.T) {
+		_, err := DecodeBtcAddress("1EYVvXLusCxtVuEwoYvWRyN5EZTXwPVvo3", BtcTestNetChain().ChainId)
```

Comment:
as far as i know the address itself does not encode network info, so this should be a valid address on all networks.